Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -1016,35 +1016,41 @@ private final class TokenStreamCreator: SyntaxVisitor {
     return .visitChildren
   }
 
-  func handleAvailabilitySpec(leftParen: TokenSyntax?, spec: Syntax, rightParen: TokenSyntax?) {
-    var tokens: [TokenSyntax] = []
-    if let leftParen = leftParen {
-      tokens.append(leftParen)
-    }
-    spec.tokens.forEach { tokens.append($0) }
-    if let rightParen = rightParen {
-      tokens.append(rightParen)
-    }
-
-    for i in 0..<(tokens.count - 1) {
-      switch (tokens[i].tokenKind, tokens[i+1].tokenKind) {
-      case (.leftParen, _): ()
-      case (_, .rightParen): ()
-      case (_, .comma): ()
-      case (_, .colon): ()
-      default:
-        after(tokens[i], tokens: .space)
-      }
+  func visit(_ node: AttributeSyntax) -> SyntaxVisitorContinueKind {
+    before(node.firstToken, tokens: .open)
+    if node.argument != nil {
+      // Wrap the attribute's arguments in their own group, so arguments stay together with a higher
+      // affinity than the overall attribute (e.g. allows a break after the opening "(" and then
+      // having the entire argument list on 1 line). Necessary spaces and breaks are added inside of
+      // the argument, using type specific visitor methods.
+      after(node.leftParen, tokens: .break(.open, size: 0), .open(argumentListConsistency()))
+      before(node.rightParen, tokens: .break(.close, size: 0), .close)
     }
+    after(node.lastToken, tokens: .close)
+    return .visitChildren
   }
 
-  func visit(_ node: AttributeSyntax) -> SyntaxVisitorContinueKind {
-    if let argument = node.argument {
-      handleAvailabilitySpec(leftParen: node.leftParen, spec: argument, rightParen: node.rightParen)
+  func visit(_ node: AvailabilitySpecListSyntax) -> SyntaxVisitorContinueKind {
+    for argument in node.dropLast() {
```

Comment:
There's a helper method defined earlier in this file named `insertTokens(_:betweenElementsOf:)` that does this, so you should be able to write:

```swift
insertTokens(.break(.same, size: 1), betweenElementsOf: node)
```

Reply:
Perfect. Done.